homeassistant/components/braviatv/remote.py

```diff
@@ -1,17 +1,29 @@
 """Remote control support for Bravia TV."""
+from __future__ import annotations
+
+from typing import Any, Iterable, cast
 
 from homeassistant.components.remote import ATTR_NUM_REPEATS, RemoteEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
 from homeassistant.helpers.update_coordinator import CoordinatorEntity
 
+from . import BraviaTVCoordinator
 from .const import ATTR_MANUFACTURER, DEFAULT_NAME, DOMAIN
 
 
-async def async_setup_entry(hass, config_entry, async_add_entities):
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
     """Set up Bravia TV Remote from a config entry."""
 
     coordinator = hass.data[DOMAIN][config_entry.entry_id]
-    unique_id = config_entry.unique_id
-    device_info = {
+    unique_id = cast(str, config_entry.unique_id)
```

Comment:
This doesn't seem right. Handle `None` value.

Reply:
In fact, `unique_id` can't be `None`, look [here](https://github.com/home-assistant/core/blob/dev/homeassistant/components/braviatv/config_flow.py#L63).

We need `str` [here](https://github.com/home-assistant/core/blob/ce66a7a85cf4d7ff32c202628f7e97ae2e2371d4/homeassistant/components/braviatv/remote.py#L27) and [here](https://github.com/home-assistant/core/blob/ce66a7a85cf4d7ff32c202628f7e97ae2e2371d4/homeassistant/components/braviatv/remote.py#L34). How can we fix this problem?

Reply:
```python
# Set in init_device
assert config_entry.unique_id is not None
```

Reply:
But `async_set_unique_id()` in `init_device()` returns `Optional[ConfigEntry]` so we get an error `Item "None" of "Optional[ConfigEntry]" has no attribute "unique_id"`.

Maybe we can use `assert unique_id is not None` [here](https://github.com/home-assistant/core/blob/aa972d77c79cf0ac7657d262ba1778ce03f00d50/homeassistant/components/braviatv/media_player.py#L53) and [here](https://github.com/home-assistant/core/blob/aa972d77c79cf0ac7657d262ba1778ce03f00d50/homeassistant/components/braviatv/remote.py#L26).

Reply:
Yes, that should work.